pkg/config/config.go

```diff
@@ -150,14 +160,35 @@ func (gc *Config) ShouldSkipTLSForAuthority(candidateAuthority string) bool {
 
 // KappDeployRawOptions returns user configured kapp raw options
 func (gc *Config) KappDeployRawOptions() []string {
-	gc.lock.RLock()
-	defer gc.lock.RUnlock()
+	gc.dataLock.RLock()
+	defer gc.dataLock.RUnlock()
 
 	// Configure kapp to keep only 5 app changes as it seems that
 	// larger number of ConfigMaps negative affects other controllers on the cluster.
 	// Eventually kapp can be smart enough to keep minimal number of app changes.
 	// Set default first so that it can be overridden by user provided options.
-	return append([]string{"--app-changes-max-to-keep=5"}, gc.kappDeployRawOptions...)
+	return append([]string{"--app-changes-max-to-keep=5"}, gc.data.kappDeployRawOptions...)
+}
+
+// AppDefaultSyncPeriod returns duration that is used by Apps
+// that do not explicitly specify sync period.
+func (gc *Config) AppDefaultSyncPeriod() time.Duration {
+	const lowestDefault = 30 * time.Second
```

Comment:
honest golang question:  does it make a difference to define this outside of the function? i would expect this kind of constant at the top of a file, from other languages.  -  are we going to allocate/evaluate this expression once per invocation,  or is there an implicit `static const` equivalent here that will optimize it out?

Reply:
> does it make a difference to define this outside of the function?

general recommendation i push for is to limit any code to their absolutely smallest scope required. pulling out consts used in small scope just creates extra indirection (which is a burden to follow through). in a perfect world most variables are consts since majority do not have change, but go does not have such level of consts.

Reply:
>  are we going to allocate/evaluate this expression once per invocation

that's up to go compiler to decide. the perf implication here is "extra" negligible.